javaparser-core/src/main/java/com/github/javaparser/TokenTypes.java

```diff
@@ -51,19 +66,23 @@ public static boolean isComment(int kind) {
     /**
      * @return the kind of EOL token to use on the platform you're running on.
      */
-    public static int eolTokenKind() {
-        if (EOL.equals("\n")) {
+    public static int eolTokenKind(LineEnding lineEnding) {
+        if (lineEnding.equalsString(LineEnding.LF)) {
             return UNIX_EOL;
         }
-        if (EOL.equals("\r\n")) {
+        if (lineEnding.equalsString(LineEnding.CRLF)) {
             return WINDOWS_EOL;
         }
-        if (EOL.equals("\r")) {
+        if (lineEnding.equalsString(LineEnding.CR)) {
             return OLD_MAC_EOL;
```

Comment:
Would it be a bad idea to put the corresponding tokens in the enum too?

Reply:
I hesitated there because `WINDOWS_EOL` AND `OLD_MAC_EOL` refer to static ints within the grammar. 

The other aspect that caused hesitation there is to keep the enum relatively "pure" and not needing to (necessarily) ever be aware of the grammar stuff.

Neither are show-stoppers though.

Reply:
Unless, of course, I'm misunderstanding you?

Reply:
Nah, I think you understand right and it's a fine decision. It made me think of this thing by the way: https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/JavaToken.java - it is my attempt to define the definitive "token" class, so that all other weird int-tokens and methods operating on them could be removed. Nothing relevant now, just info.

Reply:
Plan -- leaving as-is for now.. it can be improved in a future release.